### PR TITLE
docs(billing): add Billing section (plans, pricing, cost reduction)

### LINKED ISSUE #243: Add billing page in docs
Add a page for billing section in docs

Ref msg : [here](https://signoz-team.slack.com/archives/C02BEBF6KM2/p1718185400663599)

### PR DESCRIPTION
**Pull request by revmag**
[Staging preview](https://staging.signoz.io/docs/billing/overview/)

---

### 📄 Summary

SigNoz had no docs explaining how billing works. Pricing lived only on the marketing [pricing page](https://signoz.io/pricing/), and Cost Meter docs cover *monitoring* spend but not *how you're charged* or how plans differ. This PR adds a dedicated **Billing** docs section (issue #243), modeled on [Supabase's billing docs](https://supabase.com/docs/guides/platform/billing-on-supabase).

The section explains plans, the usage-based charging model, retention, AWS Marketplace billing, and terms/licensing. It links out to Cost Meter and the existing drop-telemetry guides instead of duplicating them, and cites the pricing page as the source of truth for exact rates.

**New Billing category** (placed at the bottom of the docs sidebar, below Community):

| Page | Route | Covers |
|------|-------|--------|
| Overview | `/docs/billing/overview` | Plans at a glance (Teams → Enterprise → Community), what each includes, startup program, AWS Marketplace pointer |
| How You're Charged | `/docs/billing/how-you-are-charged` | Usage-based model, per-signal rates + minimum retention, how logs/traces volume and metric samples are counted, retention, included usage & overages, paying through AWS Marketplace |
| Troubleshooting → FAQs | `/docs/billing/troubleshooting/faqs` | Common billing questions, security certifications (Trust Center), Terms of Service, and SigNoz dual-license explanation |

#### Issues closed by this PR

Closes #243

---

### ✅ Change Type

- [x] ✨ Feature (new documentation)

---

### 🧪 Testing Strategy

- Manual verification: reviewed all three pages on the staging preview.
- Automated checks (all passing):
  - `yarn check:docs-metadata` + `yarn test:docs-metadata`
  - `yarn check:doc-redirects` + `yarn test:doc-redirects`
  - `yarn check:stale-urls`
- Edge cases covered: all routes are new (no redirects required); link conventions verified (internal links absolute, external links use MDX anchors with `rel="noopener noreferrer nofollow"`).

---

### ⚠️ Risk & Impact Assessment

- **Blast radius:** Docs-only. New pages plus one sidebar entry; no app/runtime code touched.
- **Potential regressions:** None functional. Main risk is content accuracy of pricing figures.
- **Rollback plan:** Revert the PR; no data or config migrations involved.

---

### 📝 Changelog

| Field | Value |
|------|-------|
| Deployment Type | Cloud / OSS |
| Change Type | Feature (Documentation) |
| Description | New Billing docs section covering plans, the usage-based charging model, retention, AWS Marketplace billing, and terms/licensing. |

---

### 📋 Checklist

- [x] Tests added or explicitly not required (docs metadata/redirect/stale-url checks pass)
- [x] Manually tested (staging preview)
- [x] Breaking changes documented (none)
- [x] Backward compatibility considered (all new routes)

---

## 👀 Notes for Reviewers

Please confirm the pricing details before merge — they were sourced from the live pricing page:

- Rates: `$0.30/GB` logs & traces, `$0.10/M` metric samples; base Teams from `$49/mo`, Enterprise from `$4,000/mo`.
- **Retention pairing:** the rate table presents `$0.30/GB` (logs/traces) and `$0.10/M` (metrics) as the rate **at the minimum retention** (15 days / 1 month), with longer retention costing more. Confirm this baseline is correct.
- AWS Marketplace note says billing through AWS "can draw down on your AWS commitments" — confirm this holds for the SigNoz listing.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: data/docs/billing/troubleshooting/faqs.mdx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+Yes. The open-source, self-hosted Community Edition is free to use, with no ingestion bill. You provision and manage the infrastructure yourself. See the [Billing overview](https://signoz.io/docs/billing/overview/).
+
+## Is there a discount for startups?
+
+Eligible early-stage companies can get a discount on the Teams plan through the SigNoz startup program. See the <a href="https://signoz.io/startups/" target="_blank" rel="noopener noreferrer nofollow">SigNoz for Startups</a> page for current eligibility and details.
+
+## How is Enterprise different?
+
+Enterprise adds volume discounts, annual contracts, dedicated environments, and guided migration on top of the Teams plan. <a href="https://signoz.io/enterprise/" target="_blank" rel="noopener noreferrer nofollow">Contact the SigNoz team</a> to discuss.
```
Comment (revmag) [inline data/docs/billing/troubleshooting/faqs.mdx]: @makeavish let me know the wordings to use for explaining Enterprise plan

**Comment by makeavish** [comment]
@revmag Please always follow PR template for PR description
